src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php

```diff
@@ -165,10 +213,35 @@ protected static function createKernel(array $options = array())
             static::$class = static::getKernelClass();
         }
 
-        return new static::$class(
+        if (isset($_SERVER['KERNEL_ENV'])) {
+            $options['environment'] = $_SERVER['KERNEL_ENV'];
```

Comment:
This behavior can be useful for some kind of "integration" tests: for example you can take the external bundle and run it's test suite with your application kernel.